### PR TITLE
Refactor get model metadata

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

File: client/python/lib/ovmsclient/tfs_compat/http/serving_client.py [hunk 1]
```diff
@@ -103,21 +103,27 @@ def get_model_metadata(self, request):
             >>> response = client.get_model_metadata(request)
             >>> type(response)
         '''
-
-        HttpClient._check_model_metadata_request(request)
-
+        self._validate_timeout(timeout)
+        request = make_metadata_request(model_name, model_version)
```
Comment (mkuczyns11) [inline client/python/lib/ovmsclient/tfs_compat/http/serving_client.py:107]: I think we need to change arguments provided to `get_model_status` here. Currently we provide request, but we should provide same arguments as in grpc implementation.

File: client/python/lib/ovmsclient/tfs_compat/http/serving_client.py [hunk 2]
```diff
@@ -103,21 +103,27 @@ def get_model_metadata(self, request):
...
         raw_response = None
+
         try:
             raw_response = self.session.get(f"http://{self.url}"
                                             f"/v1/models/{request.model_name}"
                                             f"/versions/{request.model_version}"
                                             f"/metadata",
                                             cert=self.client_key, verify=self.server_cert)
```
Comment (mkuczyns11) [inline client/python/lib/ovmsclient/tfs_compat/http/serving_client.py:115]: Here we should use timeout in `self.session.get` call

**Comment by mkuczyns11** [comment]
I also think that we need to refactor some tests. For instance [here](https://github.com/openvinotoolkit/model_server/blob/09b3b9299ea09137f997609aa39e11605e0049e6/client/python/lib/tests/tfs_compat_grpc/test_grpc_serving_client.py#L319-L327) we should check whether provided model_name, model_version, timeout is correct instead of raw_request. Same for http.

File: client/python/lib/ovmsclient/tfs_compat/base/serving_client.py [hunk 3]
```diff
@@ -40,19 +40,26 @@ def predict(self, request):
...
 
         Args:
-            request: ModelMetadataRequest object.
+            model_name: name of the model in the model server.
+            model_version: version of the model (default = 0).
+            timeout: time in seconds to wait for the response (default = 10).
 
         Returns:
-            ModelMetadataResponse object
+            Dictionary with the model status response.
```
Comment (mzegla) [inline client/python/lib/ovmsclient/tfs_compat/base/serving_client.py:53]:
```suggestion
            Dictionary with the model metadata response.
```

File: client/python/lib/ovmsclient/tfs_compat/http/serving_client.py [hunk 4]
```diff
@@ -103,21 +103,27 @@ def get_model_metadata(self, request):
             >>> response = client.get_model_metadata(request)
             >>> type(response)
         '''
```
Comment (mzegla) [inline client/python/lib/ovmsclient/tfs_compat/http/serving_client.py:105]: Can you remove the above docstring? The method behavior is already described in the base class method.

**Review by mzegla — COMMENTED** [review]
Please refactor and add tests similar to the ones as in get_model_status.

File: client/python/lib/ovmsclient/tfs_compat/http/serving_client.py (hunk 1 above)
Comment (mzegla) [reply client/python/lib/ovmsclient/tfs_compat/http/serving_client.py:107]: What is wrong with this one? I don't see any issue with this part.

Comment (mkuczyns11) [reply client/python/lib/ovmsclient/tfs_compat/http/serving_client.py:107]: @mzegla highlighted part was indeed correct, however I could not highlight `get_model_status` function declaration, which was not changed and that was the problem. Everything is fine now.

File: client/python/lib/tests/tfs_compat_grpc/test_grpc_serving_client.py [hunk 5]
```diff
@@ -325,81 +326,63 @@ def test_check_model_metadata_request_invalid_type(model_metadata_request,
...
+
+    valid_grpc_serving_client_min.model_service_stub.GetModelMetadata.call_count == 0
+    assert str(error.value) == error_message
+
 
 def test_get_model_metadata_valid(mocker, valid_grpc_serving_client_min,
                                   valid_model_metadata_response):
-    model_metadata_request = mocker.Mock()
-
     mock_check_request = mocker.patch('ovmsclient.tfs_compat.grpc.serving_client'
```
Comment (mzegla) [inline client/python/lib/tests/tfs_compat_grpc/test_grpc_serving_client.py:357]: `_check_model_metadata_request` is no longer called in `get_model_metadata()`

File: client/python/lib/tests/tfs_compat_grpc/test_grpc_serving_client.py [hunk 6]
```diff
@@ -325,81 +326,63 @@ def test_check_model_metadata_request_invalid_type(model_metadata_request,
...
-                                         expected_message, grpc_error_status_code,
-                                         grpc_error_details):
-    model_metadata_request = mocker.Mock()
+                                       grpc_error_status_code, grpc_error_details,
+                                       raised_error_type, raised_error_message):
 
-    mock_check_request = mocker.patch('ovmsclient.tfs_compat.grpc.serving_client'
-                                      '.GrpcClient._check_model_metadata_request')
-    valid_grpc_serving_client_min.prediction_service_stub.GetModelMetadata\
+    valid_grpc_serving_client_min.model_service_stub.GetModelStatus\
```
Comment (mzegla) [inline client/python/lib/tests/tfs_compat_grpc/test_grpc_serving_client.py:378]: GetModelMetadata

File: client/python/lib/tests/tfs_compat_grpc/test_grpc_serving_client.py [hunk 7]
```diff
@@ -325,81 +326,63 @@ def test_check_model_metadata_request_invalid_type(model_metadata_request,
...
-    with pytest.raises(expected_exception) as e_info:
-        valid_grpc_serving_client_min.get_model_metadata(model_metadata_request)
-
-    assert str(e_info.value) == expected_message
-    assert mock_check_request.call_count == 1
+    with pytest.raises(raised_error_type) as grpc_error:
+        valid_grpc_serving_client_min.get_model_status("model_name")
 
+    assert str(grpc_error.value) == raised_error_message
+    assert valid_grpc_serving_client_min.model_service_stub.GetModelStatus.call_count == 1
```
Comment (mzegla) [inline client/python/lib/tests/tfs_compat_grpc/test_grpc_serving_client.py:385]: GetModelMetadata

**Review by mkuczyns11 — APPROVED** [review]

File: client/python/lib/tests/tfs_compat_grpc/test_grpc_serving_client.py [hunk 8]
```diff
@@ -300,36 +300,22 @@ def test_check_model_metadata_request_invalid_type(model_metadata_request,
...
-    valid_grpc_serving_client_min.model_service_stub.GetModelMetadata.call_count == 0
+    valid_grpc_serving_client_min.prediction_service_stub.GetModelMetadata.call_count == 0
     assert str(error.value) == error_message
 
 
 def test_get_model_metadata_valid(mocker, valid_grpc_serving_client_min,
                                   valid_model_metadata_response):
-    mock_check_request = mocker.patch('ovmsclient.tfs_compat.grpc.serving_client'
-                                      '.GrpcClient._check_model_metadata_request')
+    mock_check_request = mocker.patch('ovmsclient.tfs_compat.grpc.serving_client')
```
Comment (mzegla) [inline client/python/lib/tests/tfs_compat_grpc/test_grpc_serving_client.py:318]: is that line needed?

File: client/python/lib/tests/tfs_compat_grpc/test_grpc_serving_client.py [hunk 9]
```diff
@@ -300,36 +300,22 @@ def test_check_model_metadata_request_invalid_type(model_metadata_request,
...
-    (["model_name", "model_version", 10], TypeError,
-        "model_version type should be int, but is str"),
-    (["model_name", 2**63, 10], ValueError, f"model_version should be in range <0, {2**63-1}>"),
-    (["model_name", -1, 10], ValueError, f"model_version should be in range <0, {2**63-1}>"),
-    # Timeout check
-    (["model_name", 1, "string"], TypeError, "timeout value must be positive float"),
-    (["model_name", 1, 0], TypeError, "timeout value must be positive float"),
-    (["model_name", 1, -1], TypeError, "timeout value must be positive float"),
-])
+@pytest.mark.parametrize("params, expected_error, error_message", MODEL_STATUS_INVALID_PARAMS)
```
Comment (mzegla) [inline client/python/lib/tests/tfs_compat_grpc/test_grpc_serving_client.py:303]: Please use `MODEL_METADATA_INVALID_PARAMS` for consistency

File: client/python/lib/tests/tfs_compat_http/test_http_serving_client.py [hunk 10]
```diff
@@ -172,19 +172,7 @@ def test_get_model_metadata_valid(mocker, valid_http_serving_client_min,
...
-    (["model_name", "model_version", 10], TypeError,
-        "model_version type should be int, but is str"),
-    (["model_name", 2**63, 10], ValueError, f"model_version should be in range <0, {2**63-1}>"),
-    (["model_name", -1, 10], ValueError, f"model_version should be in range <0, {2**63-1}>"),
-    # Timeout check
-    (["model_name", 1, "string"], TypeError, "timeout value must be positive float"),
-    (["model_name", 1, 0], TypeError, "timeout value must be positive float"),
-    (["model_name", 1, -1], TypeError, "timeout value must be positive float"),
-])
+@pytest.mark.parametrize("params, expected_error, error_message", MODEL_STATUS_INVALID_PARAMS)
```
Comment (mzegla) [inline client/python/lib/tests/tfs_compat_http/test_http_serving_client.py:175]: Please use MODEL_METADATA_INVALID_PARAMS for consistency

File: client/python/lib/tests/tfs_compat_http/config.py [hunk 11]
```diff
@@ -410,16 +410,6 @@
 
 # (response_outputs_dict, expected_outputs_dict)
 RESPONSE_VALID_OTHER = [
```
Comment (mzegla) [inline client/python/lib/tests/tfs_compat_http/config.py:412]: This const is no longer needed when all methods have been refactored

**Review by mzegla — COMMENTED** [review]
I think there are BadResponseError cases missing. The ones with malformed server response.

**Review by mzegla — APPROVED** [review]

**Pull request merged**